Facepunch.Steamworks/Structs/LobbyQuery.cs

```diff
@@ -43,6 +43,32 @@ public LobbyQuery FilterDistanceWorldwide()
 		}
 		#endregion
 
+		#region String key/value filter
+		internal MatchMakingKeyValuePair_t? stringFilter;
+
+		/// <summary>
+		/// Filter by specified key/value pair; string parameters
+		/// </summary>
+		public LobbyQuery FilterStringKeyValue( string nKey, string nValue )
+		{
+			if ( string.IsNullOrEmpty( nKey ) )
+				throw new System.ArgumentException( "Key string provided for LobbyQuery filter is null or empty", nameof( nKey ) );
+
+			if ( nKey.Length <= SteamMatchmaking.MaxLobbyKeyLength )
```

Comment:
Sorry for being a pedant, but this is kind of a backwards way to do it. I'd throw if the argument is wrong instead of in the else.

Reply:
No worries, I appreciate your perspective and comments. I'm as much here to learn to do better as I am to do the best I can.